robottelo/ssh.py

```diff
@@ -73,18 +76,28 @@ def _get_connection(
         hostname = settings.server.hostname
     if username is None:
         username = settings.server.ssh_username
-    if key_filename is None:
+    if key_filename is None and password is None:
         key_filename = settings.server.ssh_key
+        # If we don't have key even in settings, read password instead:
+        if key_filename is None:
+            password = settings.server.ssh_password
```

Comment:
@rochacbruno exactly. At this point all the settings are already fetched, so it's just simple `foo = bar` operation, no need to overcomplicate things here. That's why i still like your initial [suggestion](https://github.com/SatelliteQE/robottelo/pull/3362/files/f8283cd2e063c80a68dc090e417bbb4bf0c3beab#r57583386) the most :)